src/main/java/com/GG/T9AgeCombat/service/ToWoundService.java

```diff
@@ -23,14 +23,15 @@ Integer rollToWound(Unit attacker, Unit defender, Integer quantity) {
         Integer toWoundThreshold = this.determineToWoundThreshold(attackerStrength, defenderToughness);
         List<Integer> dice = diceRollingService.roll(quantity);
 
-        return removeFailedToWoundRolls(dice, toWoundThreshold);
+
+        return diceRollingService.getFinalWithReRolls(dice, toWoundThreshold, attacker.getReRollToWoundLessThan(), attacker.getReRollToWoundGreaterThan());
     }
 
     Integer determineToWoundThreshold(Integer attackerStrength, Integer defenderToughness) {
-        return TO_WOUND_DEFAULT_THRESHOLD - (attackerStrength - defenderToughness);
-    }
+        Integer preliminaryToWoundThreshold = TO_WOUND_DEFAULT_THRESHOLD - (attackerStrength - defenderToughness);
 
-    Integer removeFailedToWoundRolls(List<Integer> resultList, Integer toWoundThreshold) {
-        return (int) resultList.stream().filter(a -> (a != TO_WOUND_AUTO_FAIL && a >= toWoundThreshold) || a == TO_WOUND_AUTO_SUCCESS).count();
+        return  preliminaryToWoundThreshold <= TO_WOUND_AUTO_FAIL ?
+                TO_WOUND_AUTO_FAIL + 1 : preliminaryToWoundThreshold > TO_WOUND_AUTO_SUCCESS ?
```

Comment:
Let's define another constant for TO_WOUND_AUTO_FAIL + 1 like MINIMUM_TO_WOUND_THRESHOLD